src/main/java/jalf/aggregator/Avg.java

```diff
@@ -0,0 +1,72 @@
+package jalf.aggregator;
+
+import jalf.AttrName;
+import jalf.Tuple;
+
+public class  Avg implements Aggregator<Double>{
+    private AttrName aggregatedField;
+    private  Double state;
+    private  Integer counttuple ;
+
+
+
+    public static Avg avg(AttrName attr) {
+        return new Avg(attr);
+    }
+
+
+    public Avg(AttrName aggregatedField) {
+        super();
+        this.setAggregatedField(aggregatedField);
+        this.init();
+    }
+
+
+
+    public void setAggregatedField(AttrName aggregatedField) {
+        this.aggregatedField= aggregatedField;
+    }
+
+
+
+
+    @Override
+    public Double finish() {
+        if (this.counttuple > 0 ){
+            this.state =this.state/this.counttuple;
+        }
+        else {
+            this.state=0.0;
+        }
+        return this.state;
+    }
+
+    @Override
+    public void init() {
+        this.state=new Double(0);
+        this.counttuple=new Integer(0);
+
+    }
+
+    @Override
+    public void accumulate(Tuple t) {
+        // ajouter les autres types
+        Integer value=(Integer) t.get(this.getAggregatedField());
+        this.state= this.state.doubleValue() +value;
+        this.counttuple=this.counttuple+1;
+
+    }
+
+
+    public AttrName getAggregatedField() {
+        return this.aggregatedField;
+    }
+
+
+    public Avg  conclude(Avg other) {
+        this.counttuple=other.counttuple;
+        this.state = other.state;
+        return this;
+    }
```

Comment:
conclude est une fonction appeler dans la méthode test de Summarize.java .
map = tuples.collect(Collectors.groupingBy(t -> t.fetch(byNameAttrs),
                    Collector.of(Count::new, Count::accumulate, Count::conclude)));

On utilise la méthode of de Collector qui utilise la méthode conclude pour récupere l'aggregator résultant. En soit, of prend un nouvelle aggregator, un accumulateur qui fait le calcul et enfin d'une méthode de terminaison (conclude dans ce cas) qui prend en paramettre l'aggregator et renvoi une valeur (le même aggregator).

On va la rajouter dans l'interface.